src/App.jsx

```diff
@@ -1,12 +1,117 @@
 import './App.css';
+import {useState} from "react";
+import axios from "axios";
+import regionColor from "./helpers/regionColor.js";
+import './helpers/regionColor.css';
+import worldMap from './assets/world_map.png'
+import globe from './assets/globe.png'
+import formatToMillions from "./helpers/formatToMillions.js";
+
 
 function App() {
 
+
+    const [countries, setCountries] = useState([]);
+    const [error, setError] = useState('');
+    const [countryInformation, setCountryInformation] = useState([]);
+    const [searchQuery, setSearchQuery] = useState('');
+
+
+    async function fetchAllCountries() {
+        try {
+            setError('');
+            const response = await axios.get('https://restcountries.com/v3.1/all');
+            const sortedCountries = response.data.sort((a, b) => {
+                return (a.population) - (b.population);
+            });
+            setCountries(sortedCountries);
+
+        } catch (e) {
+            console.error(e);
+            setError('Het ophalen van de data is mislukt. Probeer het opnieuw');
+        }
+    }
+
+
+    async function fetchCountryInformation() {
+        setError('');
+        try {
+            const response = await axios.get(`https://restcountries.com/v3.1/name/${searchQuery}?fullText=true`);
+            console.log(response.data)
+            setCountryInformation(response.data);
+            setSearchQuery('');
+        } catch (e) {
+            console.error(e);
+            setError(`${searchQuery} bestaat niet. Probeer het opnieuw`);
+        }
+    }
+
+
+    const handleInputChange = (event) => {
+        setSearchQuery(event.target.value);
+    };
+
+    const handleSearch = (e) => {
+        e.preventDefault();
+        fetchCountryInformation();
+    };
+
+
+
     return (
         <>
-            Maak je applicatie hier!
+
+            <img className="worldMap" src={worldMap} alt="world-map-img"/>
+            <h1>World Regions</h1>
+            <button type="submit" onClick={fetchAllCountries}>
+                Get info
+            </button>
+            {error && <p className="error">{error}</p>}
+
+
+            {countries.length > 0 && (
+                <ul>
+                    {countries.map((country, index) => (
```

Comment:
Je kunt hier nog spreading toepassen, zodat je niet overal `country.` vóór hoeft te zetten
```suggestion
                    {countries.map(({ name, region, flags, population }, index) => (
```